jvm/src/main/java/com/microsoft/bond/protocol/CompactBinaryProtocolWriter.java

```diff
@@ -0,0 +1,383 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+import com.microsoft.bond.Metadata;
+import com.microsoft.bond.ProtocolType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Stack;
+
+// TODO: add Compact Binary protocol format comment (either here and/or in the deserializer and/or in package.html)
+
+/**
+ * Implements Compact Binary protocol (serialization).
+ */
+public final class CompactBinaryProtocolWriter implements TwoPassProtocolWriter {
```

Comment:
Interface is named "ProtocolWriter" and it's usual to have interface name in the suffix of the implementation.

---
In reply to: [116114672](https://github.com/tstein/bond/pull/4#discussion_r116114672) [](ancestors = 116114672)